cgsmiles/read_fragments.py

```diff
@@ -41,7 +72,14 @@ def strip_bonding_descriptors(fragment_string):
                 while peek != ']':
                     bond_descrp += peek
                     peek = next(smile_iter)
-                bonding_descrpt[prev_node].append(bond_descrp)
+                if smile_iter.peek() in bond_to_order:
+                    order = bond_to_order[next(smile_iter)]
+                elif current_order:
+                    order = current_order
+                    current_order = None
+                else:
+                    order = 1
+                bonding_descrpt[prev_node].append(bond_descrp + str(order))
```

Comment:
Not 100% sure I understand what's happening here.

Isn't it easier to do:
`bonding_descrpt[prev_node].append({'type': bond_descrp[0], 'id': bond_descrp[1:], 'order': order})`